cli/cmd/release_create.go

```diff
@@ -7,68 +7,77 @@ import (
 	"github.com/spf13/cobra"
 )
 
-var createReleaseYaml string
-var createReleasePromote string
-var createReleasePromoteRequired bool
-var createReleasePromoteNotes string
-var createReleasePromoteVersion string
-
-var releaseCreateCmd = &cobra.Command{
-	Use:   "create",
-	Short: "Create a new release",
-	Long: `Create a new release by providing YAML configuration for the next release in
-your sequence.`,
-}
+func (r *runners) InitReleaseCreate(parent *cobra.Command) {
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Create a new release",
+		Long: `Create a new release by providing YAML configuration for the next release in
+  your sequence.`,
+	}
+
+	parent.AddCommand(cmd)
 
-func init() {
-	releaseCmd.AddCommand(releaseCreateCmd)
+	cmd.Flags().StringVar(&r.args.createReleaseYaml, "yaml", "", "The YAML config for this release. Use '-' to read from stdin.  Cannot be used with the `yaml-file` falg.")
+	cmd.Flags().StringVar(&r.args.createReleaseYamlFile, "yaml-file", "", "The file name with YAML config for this release.  Cannot be used with the `yaml` flag.")
+	cmd.Flags().StringVar(&r.args.createReleasePromote, "promote", "", "Channel name or id to promote this release to")
+	cmd.Flags().StringVar(&r.args.createReleasePromoteNotes, "release-notes", "", "When used with --promote <channel>, sets the **markdown** release notes")
+	cmd.Flags().BoolVar(&r.args.createReleasePromoteRequired, "required", false, "When used with --promote <channel>, marks this release as required during upgrades.")
+	cmd.Flags().StringVar(&r.args.createReleasePromoteVersion, "version", "", "When used with --promote <channel>, sets the version label for the release in this channel")
 
-	releaseCreateCmd.Flags().StringVar(&createReleaseYaml, "yaml", "", "The YAML config for this release. Use '-' to read from stdin")
-	releaseCreateCmd.Flags().StringVar(&createReleasePromote, "promote", "", "Channel name or id to promote this release to")
-	releaseCreateCmd.Flags().StringVar(&createReleasePromoteNotes, "release-notes", "", "When used with --promote <channel>, sets the **markdown** release notes")
-	releaseCreateCmd.Flags().BoolVar(&createReleasePromoteRequired, "required", false, "When used with --promote <channel>, marks this release as required during upgrades.")
-	releaseCreateCmd.Flags().StringVar(&createReleasePromoteVersion, "version", "", "When used with --promote <channel>, sets the version label for the release in this channel")
+	cmd.RunE = r.releaseCreate
 }
 
 func (r *runners) releaseCreate(cmd *cobra.Command, args []string) error {
-	if createReleaseYaml == "" {
+	if r.args.createReleaseYaml == "" && r.args.createReleaseYamlFile == "" {
 		return fmt.Errorf("yaml is required")
 	}
 
-	if createReleaseYaml == "-" {
+	if r.args.createReleaseYaml != "" && r.args.createReleaseYamlFile != "" {
+		return fmt.Errorf("only yaml or yaml-file has to be specified")
```

Comment:
mind if we re-word this a little bit?
```suggestion
		return fmt.Errorf("only one of yaml or yaml-file may be specified")
```